src/install-script/installer.ts

```diff
@@ -0,0 +1,200 @@
+import * as archiver from "archiver";
+import { ChildProcess, exec, spawn } from "child_process";
+import { createWriteStream, ensureDir, remove } from "fs-extra";
+import { join as joinPath } from "path";
+
+export enum ScriptMessageType {
+  MINOR,
+  MAJOR,
+  ERROR,
+  COMPLETE
+}
+
+export enum InfoSource {
+  NPM = "npm",
+  VERDACCIO = "verdaccio",
+  OFFLINE_NPM = "offline-npm",
+  ARCHIVER = "archiver"
+}
+
+export interface IScriptMessage {
+  type: ScriptMessageType;
+  message: string;
+  source: InfoSource;
+}
+
+// Utility to make using callbacks with IScriptMessages easier
+const makeOutputter = (
+  callback: (data: IScriptMessage) => void,
+  type: ScriptMessageType
+) => (message: string, source: InfoSource = InfoSource.OFFLINE_NPM) => {
+  callback({
+    message,
+    source,
+    type
+  });
+};
+
+/**
+ * Installs an npm cache through Verdaccio and bundles the results in a zip file.
+ * @param packages the list of package names to install
+ * @param onInfo callback function for info messages
+ * @param onError callback function for error messages
+ * @param onComplete callback function for when the installation is complete
+ * @param outputPath the directory in which storage.zip should be placed
+ * @param verdaccioPath the directory in which the verdaccio source is located (usually node_modules)
+ */
+export const installCache = (
```

Comment:
This is quite a few parameters. It's really easy to mess up the order for them, and is a bit unclear when passing to it. It's also harder to make some optional.

Instead, how about using an interface?

```typescript
interface InstallCacheSettings {
    packages: string[];
    onInfo: (data: IScriptMessage) => void;
    verdaccioPath?: string;
    // ...and the other things
}
```

You can then do fun defaulting behavior like:

```typescript
export const installCache = ({
    packages,
    onInfo,
    verdaccioPath = joinPath(...)
}: InstallCacheSettings) => { ... }
```